### PR TITLE
Archives styling and functionality, leaderboard styling

### PR DESCRIPTION
**Pull request by tehnalogos**

### CODE REVIEW

File: frontend/components/Archives.tsx [hunk 1]
```diff
@@ -42,186 +39,232 @@ export interface IFetchArchives {
...
     ownerProfiles: { [key: string]: any },
     setOwnerProfiles: React.Dispatch<
       React.SetStateAction<{ [key: string]: any }>
     >,
   ): Promise<void>;
 }
 
+export interface IFetchArchivesCount {
+  (
+    setArchivesCount: React.Dispatch<React.SetStateAction<number | undefined>>,
```
Comment (democrazyfi) [inline frontend/components/Archives.tsx:52]: can this be undefined? it seems there will always be a number

File: frontend/components/Archives.tsx [hunk 2]
```diff
@@ -42,186 +39,232 @@ export interface IFetchArchives {
...
-      setLoadedIndices,
-      ownerProfiles,
-      setOwnerProfiles,
-    );
-  }, [fetchArchives, slideAmount, refineEventCounter]);
+    fetchArchivesCount(setArchivesCount);
+  }, [fetchArchivesCount, refineEventCounter]);
+
+  useEffect(() => {
+    slideAmount &&
```
Comment (democrazyfi) [inline frontend/components/Archives.tsx:109]: is this conditional required?

File: frontend/components/Archives.tsx [hunk 3]
```diff
@@ -42,186 +39,232 @@ export interface IFetchArchives {
...
 
 interface ArchivesProps {
   fetchArchives: IFetchArchives;
+  fetchArchivesCount: IFetchArchivesCount;
 }
 
-const Archives: React.FC<ArchivesProps> = ({ fetchArchives }) => {
+const Archives: React.FC<ArchivesProps> = ({
+  fetchArchivesCount,
+  fetchArchives,
```
Comment (democrazyfi) [inline frontend/components/Archives.tsx:67]: I feel we're reaching a point where re-using this Archives component for both archives and contributions is reaching a point where maybe splitting them and having a bit of duplication is better than this state

They have different way to fetch data, and they display different items - and it's hard to follow what the component because the logic is spread around

Initially we only intended to have one carousel but If our aim is to re-use carousel featues, we may his a Carousel component which renders "things" and pass result/data down

File: frontend/utils/universalProfile.ts [hunk 4]
```diff
@@ -41,6 +37,28 @@ export const getProfileData = async (
     .LSP3Profile as LSP3ProfileMetadata;
 };
 
+export const getProfileBasicInfo = async (
+  contributor: string,
+  rpcUrl: string,
+): Promise<{ upName: string | null; avatar: string | null } | null> => {
```
Comment (democrazyfi) [inline frontend/utils/universalProfile.ts:43]:
```suggestion
): Promise<{ upName: string | null; avatar: string | null }> => {
```
minor, this no longer returns null object

File: frontend/components/Leaderboard.tsx [hunk 5]
```diff
@@ -37,25 +43,37 @@ const Leaderboard: React.FC = () => {
...
+          if (Number(contribution) === 0) {
+            zeroContributionIndeces.push(index);
+          }
+        });
+        topContributors = topContributors.filter(
+          (_, index) => !zeroContributionIndeces.includes(index),
+        );
+        contributions = contributions.filter(
+          (_, index) => !zeroContributionIndeces.includes(index),
+        );
```
Comment (democrazyfi) [inline frontend/components/Leaderboard.tsx:62]: Looking at https://github.com/yearone-io/burntpix-archives/blob/main/solidity/contracts/BurntPixArchives.sol#L230

I'm not quite sure how we can end up with contributors that have zero contributions

But if there is such case, would we be able to not return those from the smart contract? so that we don't have to do this extra logic here

File: frontend/components/Archives.tsx (hunk 1 above)
Comment (tehnalogos) [reply frontend/components/Archives.tsx:52]: @democrazyfi starts off undefined. that way we cant easily track that result is still loading

File: frontend/components/Leaderboard.tsx (hunk 5 above)
Comment (tehnalogos) [reply frontend/components/Leaderboard.tsx:62]: @democrazyfi when there not enough spots to fill 10. smart contract by default returns zero address w/ zero contributions for those slots

File: frontend/components/Archives.tsx (hunk 3 above)
Comment (tehnalogos) [reply frontend/components/Archives.tsx:67]: @democrazyfi realize that things have changed from original, but core logic is pretty straightforward. i can explain on call. feel like reworking this now would take a while. how data is fetched feels like should be outside of carousel component logic anyways. whereas carousel purely takes care of rendering and knowing when to trigger fetching next batch of data

File: frontend/components/Archives.tsx (hunk 2 above)
Comment (tehnalogos) [reply frontend/components/Archives.tsx:109]: yes because initially slide amount is 0 until page determines correct value. so when it's 0 doesnt make sense to fetch anything because we dont know how much to fetch

**Review by Ledespana — APPROVED** [review]
